src/main.ts

```diff
@@ -64,6 +64,10 @@ async function main() {
             await executeAzCliCommand(`cloud set -n "${environment}"`, false);
             console.log(`Done registering cloud: "${environment}"`);
         }
+            //setting context to Azure Gov Cloud
+        if (environment.toLowerCase() =='azureusgovernment' || environment.toLowerCase() =='azuregermancloud' || environment.toLowerCase() =='azurechinacloud' || environment.toLowerCase() =='azurecloud' ) {

```

Comment:
can we make an enum or set with all government cloud values and check environment variable?

Reply:
+ Add validation for supported values of environment input

Reply:
+1 for this one

Reply:
@asranja @ammohant If user registers their own cloud in AzureStack (say MyOwnCloud) and if we validate for environment variable supported values  which would be AzureUSGovernment, AzureChinaCloud,AzureGermanCloud,AzureStack or AzureCloud  can that creates failures in AzureStack.For Sovereign cloud it would be AzureUSGovernment, AzureChinaCloud,AzureGermanCloud or AzureCloud. But it seems AzureStack gives option of registering user Cloud with a custom name.